src/types/type_wrappers.rs

```diff
@@ -69,6 +70,51 @@ impl ComplexType {
         // Batch query to NAPI minimizes number of calls
         self.inner_types.clone()
     }
+
+    /// Create a new copy of the ComplexType, with first support type set to the provided type and the same base type
+    ///
+    /// Intended for filling types of the list / set values in case no support type was initially provided
+    #[napi]
+    pub fn remap_list_support_type(&self, new_subtype: &ComplexType) -> ComplexType {
+        ComplexType::one_support(self.base_type, Some(new_subtype.clone()))
+    }
+
+    #[napi]
+    /// Create a new copy of the ComplexType, with first and second support type set to the provided types and the same base type
+    ///
+    /// Intended for filling types of the map keys and value in case no support types were initially provided
+    pub fn remap_map_support_type(
+        &self,
+        key_new_subtype: &ComplexType,
+        val_new_subtype: &ComplexType,
+    ) -> ComplexType {
+        ComplexType::two_support(
+            self.base_type,
+            Some(key_new_subtype.clone()),
+            Some(val_new_subtype.clone()),
+        )
+    }
+
+    /// Create a new ComplexType for tuple with provided inner types.
+    #[napi]
+    pub fn remap_tuple_support_type(new_subtypes: Vec<Option<&ComplexType>>) -> ComplexType {
+        ComplexType::from_tuple(
+            new_subtypes
+                .into_iter()
+                // Temporary hack:
+                // There is a chance, user doesn't provide a type for some tuple value.
+                // If this vale is null or unset, we can still correctly handle that case.
+                // For this reason we set here Unset type, a type that will never be used in request.
+                // If we encounter Unset in parsing value, this means, that unsufficient type information was provided.
+                //
+                // This will be fixed at a later time, as it requires more investigation into how tuple works.
+                .map(|e| {
+                    e.unwrap_or(&ComplexType::simple_type(CqlType::Unprovided))
+                        .clone()
+                })
+                .collect(),
```

Comment:
```suggestion
                // HACK:
                // There is a chance, user doesn't provide a type for some tuple value.
                // If this vale is null or unset, we can still correctly handle that case.
                // For this reason we set here Unprovided type, a type that will never be used in request.
                // If we encounter Unprovided in parsing value, this means, that unsufficient type information was provided.
                //
                // This will be fixed at a later time, as it requires more investigation into how tuple works.
                .map(|e| {
                    e.unwrap_or(&ComplexType::simple_type(CqlType::Unprovided))
                        .clone()
                })
                .collect(),
```